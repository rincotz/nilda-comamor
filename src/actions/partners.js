import db from '../firebase/firebase'
import { ADD_PARTNER, EDIT_PARTNER, REMOVE_PARTNER } from "./constants";

// ADD_PARTNER
export const addPartner = (partner) => ({
    type: ADD_PARTNER,
    partner
})

export const startAddPartner = (partnerData = {}) => {
    return (dispatch) => {
        const {
            uid,
            id,
            name,
            gender,
            birth,
            email,
            phone,
            bio,
            nationality,
            addressLine1,
            addressLine2 = '',
            number,
            neighborhood,
            zip,
            picture,
            geoPoint,
            guestRatings = '',
            cookerRatings = '',
            ratings = ''
        } = partnerData
        const partner = {
            uid,
            id,
            name,
            gender,
            birth,
            email,
            phone,
            bio,
            nationality,
            addressLine1,
            addressLine2,
            number,
            neighborhood,
            zip,
            picture,
            geoPoint,
            guestRatings,
            cookerRatings,
            ratings
        }
        return db.collection('partners')
            .doc(partner.uid)
            .set(partner)
            .then((docref) => {
                console.log(`Sucessfully added partner ${partner.uid}`)
        })
    }
}

//EDIT_PARTNER
export const editPartner = (uid, updates) => ({
    type: EDIT_PARTNER,
    uid,
    updates
})

export const startEditPartner = (uid, updates) => {
    return (dispatch) => {
        return db.collection('partners').doc(uid).update(updates)
            .then(() => {
                dispatch(editPartner(uid, updates))
            })
    }
}

//REMOVE_PARTNER
export const removePartner = ({ uid }) => ({
    type: REMOVE_PARTNER,
    uid
})

export const startRemovePartner = ({ uid }) => {
    return (dispatch) => {
        return db.collection('partners').doc(uid).delete()
            .then(dispatch(removePartner({ uid })))
    }
}

//SET_PARTNER
export const startSetPartner = (partnerUID) => {
    return (dispatch) => {
        return db.collection('partners').doc(partnerUID)
            .get()
            .then((partner) => {
                dispatch(addPartner(partner.data()))
            })
    }
}