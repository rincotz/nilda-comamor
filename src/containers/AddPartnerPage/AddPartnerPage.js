import React from 'react';
import { connect } from 'react-redux'
import PartnerForm from "../PartnerForm/PartnerForm";
import { startAddPartner } from "../../actions/partners";

import FormLabel from '@material-ui/core/FormLabel'

export class AddPartnerPage extends React.Component {
    onSubmit = (partner) => {
        this.props.startAddPartner(partner)
        this.props.history.push('/')
    }
    render() {
        return (
            <div>
                <FormLabel component='legend'>Complete seu cadastro</FormLabel>
                <PartnerForm
                    onSubmit={this.onSubmit}
                />
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startAddPartner: (partner) => dispatch(startAddPartner(partner))
})

export default connect(undefined, mapDispatchToProps)(AddPartnerPage);