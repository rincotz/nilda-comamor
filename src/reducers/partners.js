import {ADD_PARTNER, EDIT_PARTNER, REMOVE_PARTNER } from "../actions/constants"

// Partner reducer
export const partnerReducerDefaultState = {
    uid: "",
    id: "",
    name: "",
    gender: "",
    birth: "",
    email: "",
    phone: "",
    bio: "",
    nationality: "",
    addressLine1: "",
    addressLine2: "",
    number: "",
    neighborhood: "",
    zip: "",
    picture: "",
    geoPoint: "",
    guestRatings: "",
    cookerRatings: "",
    ratings: ""
}
export default (state = partnerReducerDefaultState, action) => {
    switch (action.type) {
        case ADD_PARTNER:
            return action.partner
        case REMOVE_PARTNER:
            return partnerReducerDefaultState
        case EDIT_PARTNER:
            return {
                ...partner,
                ...action.updates
            }
        default:
            return state
    }
}