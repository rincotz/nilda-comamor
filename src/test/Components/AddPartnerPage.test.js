import React from 'react'
import { shallow } from 'enzyme/build'
import { AddPartnerPage } from "../../containers/AddPartnerPage/AddPartnerPage"
import partners from '../fixtures/partners'

let startAddPartner, history, wrapper

beforeEach(() => {
    startAddPartner = jest.fn()
    history = { push: jest.fn() }
    wrapper = shallow(<AddPartnerPage startAddPartner={startAddPartner} history={history} />)
})

test('should render AddPartnerPage correctly', () => {
    expect(wrapper).toMatchSnapshot()
})

test('should handle onSubmit', () => {
    wrapper.find('PartnerForm').prop('onSubmit')(partners[1])
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(startAddPartner).toHaveBeenLastCalledWith(partners[1])
})