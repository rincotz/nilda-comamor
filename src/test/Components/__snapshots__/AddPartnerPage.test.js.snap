// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddPartnerPage correctly 1`] = `
<div>
  <h1>
    Complete seu cadastro
  </h1>
  <PartnerForm
    onSubmit={[Function]}
  />
</div>
`;
